filters.py

```diff
@@ -0,0 +1,154 @@
+                return
+            else:
+                value = " ".join(args[1::])
+                msg_to_log = await self._db.store_asset(value)
+        else:
+            value = reply.text
+            msg_to_log = await self._db.store_asset(reply)
+        filters[chatid][name] = str(msg_to_log)
+        self._db.set("FilterModule", "filters", filters)
+        await message.edit("<b>Text ''{}'' has been successfully filtered in this chat.</b>".format(name))
+
+    async def stopcmd(self, message):
+        """Removes a filter from the list."""
+        filtern = str(utils.get_args_raw(message))
+        filters = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        if not filtern:
+            await message.edit("<b>Please specify the name of the filter.</b>")
+            return
+        try:
+            del filters[chatid][filtern]
+            await message.edit("<b>Filter ''{}'' successfully removed from the chat.</b>".format(filtern))
+            self._db.set("FilterModule", "filters", filters)
+        except Exception:
+            await message.edit("<b>Filter ''{}'' not found in this chat</b>".format(filtern))
+
+    async def stopallcmd(self, message):
+        """Clears out the filter list."""
+        filters = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            del filters[chatid]
+            self._db.set("FilterModule", "filters", filters)
+            await message.edit("<b>All filters successfully removed from the chat.</b>")
+        except Exception:
+            await message.edit("<b>There are no filters to clear out in this chat.</b>")
+
+    async def filterscmd(self, message):
+        """Shows saved filters."""
+        filters = ""
+        filt = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            for i in filt[chatid]:
+                filters += "<b> -  " + str(i) + "</b>\n"
+                pass
+        except Exception:
+            pass
+        filterl = "<b>Text that you filtered in this chat: </b>\n\n" + filters
+        if filters:
+            await message.edit(filterl)
+        else:
+            await message.edit("<b>No filters found in this chat.</b>")
+
+    async def watcher(self, message):
+        # This is to execute cmds
+        exec = True
+        for load in self.allloaders:
+            if load.client is self._client:
+                break
+        args = str(message.text.lower())
+        filters = self._db.get("FilterModule", "filters")
+        chatid = str(message.chat_id)
+        try:
+            current_prefix = self._db.get("friendly-telegram.main", "command_prefix")
+        except Exception:
+            current_prefix = "."
+        if filters and chatid in filters:
+            for key in filters[chatid]:
+                if key.lower() in args.lower():
```

Comment:
It might be better to lower() them at save time, to optimise it and avoid lower()ing in a loop